changelog/unreleased/grpc-tls-settings.md

```diff
@@ -0,0 +1,42 @@
+Enhancement: Allow to enable TLS for grpc service
+
+We added new configuration settings for the grpc based services allowing to enable
+transport security for the services. By setting:
+
+```toml
+[grpc.tls_settings]
+enabled = true
+certificate = "<path/to/cert.pem>"
+key = "<path/to/key.pem>"
+```
+
+TLS transportsecurity is enabled using the supplied certificate. When `enabled` is set
+to `true`, but no certificate and key files are supplied reva will generate
+temporary self-signed certificates at startup (this requires to also configure
+the clients to disable certificate verification, see below).
+
+The client side can be configured via the shared section. Set this to configure the CA for
+verifying server certificates:
+
+```toml
+[shared.grpc_client_options]
+tls_mode = "on"
```

Comment:
@micbar @C0rby Different to my initial approach, where I just had a `tls_insecure` boolean this  is introducing a third state for the client's tls setting which allows to completely turn off TLS (e.g. for debugging, or for isolated deployments where TLS is not really needed).

This is somewhat different to the way we configure TLS clients in other places, but I think we need this kind of switch. Are you ok with this, or I go back to the `tls_insecure=true/false` setting without allowing to turn TLS off.

Reply:
Yeah, I think it could make sense to turn TLS off in certain situations.
This means we need to adjust those other places but we can do that in separate PRs.